plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java

```diff
@@ -139,6 +150,32 @@ public Object buildAll(String modelName, Project rootProject) {
     return new DefaultGradleSourceSets(gradleSourceSets);
   }
 
+  private String stripPathPrefix(String projectPath) {
+    if (projectPath.startsWith(":"))
+      return projectPath.substring(1);
+    return projectPath;
+  }
+
+  private String createUniqueDisplayName(DefaultGradleSourceSet sourceSet,
+    Map<SourceSet, DefaultGradleSourceSet> sourceSetMap) {
+
+    String fullName = stripPathPrefix(sourceSet.getProjectPath()) + "-" + sourceSet.getSourceSetName();
+    // has the suffix caused a clash - apply a numbered suffix
+    String usedName = fullName;
+    int i = 2;
+    boolean found = true;
+    while (found) {
+      String nameToTest = usedName;
+      if (!sourceSetMap.values().stream().anyMatch(ss -> nameToTest.equals(ss.getDisplayName())))
```

Comment:
Let's use `noneMatch` instead of negated `anyMatch`

Reply:
Done